tests/integration/test_task.py

```diff
@@ -80,3 +83,245 @@ def _run(self, input_files):
+        script=diag_script,
+        settings=diag_settings,
+        output_dir=diag_output_dir,
+    )
+    tasks = {task}
+
+    return tasks
+
+
+def test_py_diagnostic_task_basics(tmp_path):
+    """Test DiagnosticTask basic attributes."""
+    diag_script = tmp_path / 'diag_cow.py'
+    tasks = _get_single_diagnostic_task(tmp_path, diag_script)
+    task_names = [task.name for task in tasks]
+    assert task_names == ['task0']
+    ancestor_names = [anc.name for anc in list(tasks)[0].ancestors]
+    assert ancestor_names == ['task0-ancestor0', 'task0-ancestor1']
+    task = list(tasks)[0]
+    assert task.script == diag_script
+    assert task.settings == {'run_dir': tmp_path / 'mydiag' / 'run_dir',
+                             'profile_diagnostic': False}
+    assert task.output_dir == tmp_path / 'mydiag'
+
+
+def test_diagnostic_diag_script_none(tmp_path):
+    """Test case when diagnostic script doesnt exist."""
+    diag_script = tmp_path / 'diag_cow.py'
+    with pytest.raises(DiagnosticError) as err_msg:
+        _get_single_diagnostic_task(tmp_path, diag_script, write_diag=False)
+    diagnostics_root = os.path.join(DIAGNOSTICS_PATH, 'diag_scripts')
+    script_file = os.path.abspath(os.path.join(diagnostics_root, diag_script))
+    suf = "Cannot execute script "
+    ept = suf + "'{}' ({}): file does not exist.".format(script_file,
+                                                         script_file)
+    assert ept == str(err_msg.value)
+
+
+def _get_diagnostic_tasks(tmp_path, diagnostic_text, extension):
+    """Assemble Python diagnostic tasks of DiagnosticTasks."""
+    diag = 'diag_cow.' + extension
+    diag_script = tmp_path / diag
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_plot_dir = diag_output_dir / 'plot_dir'
+    diag_work_dir = diag_output_dir / 'work_dir'
+    diag_settings = {'run_dir': diag_run_dir.as_posix(),
+                     'plot_dir': diag_plot_dir.as_posix(),
+                     'work_dir': diag_work_dir.as_posix(),
+                     'profile_diagnostic': False,
+                     'exit_on_ncl_warning': False}
+
+    with open(diag_script, "w") as fil:
+        fil.write(diagnostic_text)
+
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=None,
+        script=diag_script.as_posix(),
+        settings=diag_settings,
+        output_dir=diag_output_dir.as_posix(),
+    )
+
+    return {task}
+
+
+def test_py_diagnostic_run_sequential_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask sequentially with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_sequential(tasks)
```

Comment:
It would be better if you just used the `task.run()` method instead of the `_run_tasks_sequential` and `_run_tasks_parallel` methods, because these are already tested by `test_run_tasks`. It's better to test smaller amounts of code, because that 1) makes it easier to find where the bug is if a test fails and 2) requires fewer changes to the tests if you change the code.

Reply:
done! good call, B-man

Reply:
Thanks, did you `git push` the changes? I cannot see them yet.

Reply:
ya, take off your sunglasses :sunglasses:

Reply:
Now I see it, you did it everywhere except above this comment ;-)

Reply:
:facepalm: